client/trino-jdbc/src/main/java/io/trino/jdbc/TrinoConnection.java

```diff
@@ -233,7 +241,8 @@ public void close()
                     }
                 }
                 if (transactionId.get() != null) {
-                    try (TrinoStatement statement = new TrinoStatement(this)) {
+                    try (TrinoStatement statement = new TrinoStatement(this, this::unregisterStatement)) {
+                        registerStatement(statement);
```

Comment:
leverage `createStatement()` here

Reply:
`createStatement()` returns `javax.sql.Statement` so I would need casting or method extraction.

Reply:
both sound better to me than having to remember about register & unregister

btw does the type of `TrinoStatement` matter here?